rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -555,20 +555,25 @@ def decode_transaction_hashes(
                 tx_hashes = [EVMTxHash(x[0]) for x in cursor]
 
         for tx_hash in tx_hashes:
-            try:
-                receipt = self.transactions.get_or_query_transaction_receipt(tx_hash)
-            except RemoteError as e:
-                raise InputError(f'{self.evm_inquirer.chain_name} hash {tx_hash.hex()} does not correspond to a transaction') from e  # noqa: E501
-
             # TODO: Change this if transaction filter query can accept multiple hashes
-            with self.database.conn.read_ctx() as cursor:
-                txs = self.dbevmtx.get_evm_transactions(
-                    cursor=cursor,
-                    filter_=EvmTransactionsFilterQuery.make(tx_hash=tx_hash, chain_id=self.evm_inquirer.chain_id),  # noqa: E501
-                    has_premium=True,  # ignore limiting here
+            with self.database.conn.read_ctx() as read_cursor:
+                tx = self.dbevmtx.get_or_create_transaction(
+                    cursor=read_cursor,
+                    evm_inquirer=self.evm_inquirer,
+                    tx_hash=tx_hash,
+                    relevant_address=None,
+                )
+                receipt = self.dbevmtx.get_receipt(
+                    cursor=read_cursor,
+                    tx_hash=tx_hash,
+                    chain_id=self.evm_inquirer.chain_id,
+                )
+            if receipt is None:
+                raise DeserializationError(
+                    f'{self.evm_inquirer.chain_name} hash {tx_hash.hex()} does not have a corresponding receipt in the database',  # noqa: E501
                 )
```

Comment:
The receipt is needed by the _get_or_decode_transaction_events which is called next. The check is indeed not needed becasue the recipt will not be None but mypy compains. I can just type ignore the warning instead.

Reply:
Ofc you should type ignore the comment. We don't add extra runtime code just to make mypy happy. Explain why it should exist there.

What's more since get_or_create_transaction also returns the receipt internally just make it return tuple of transaction/receipt.

Also make `assert_tx_data_is_pulled` return them too. I just noticed that in `get_or_create_transaction` you are actually requerying the DB to get the transaction which is silly